api/src/identity-access-management/domain/usecases/authenticate-user.js

```diff
@@ -49,11 +50,13 @@ const authenticateUser = async function ({
 
     const userLogin = await userLoginRepository.findByUserId(foundUser.id);
     if (foundUser.email && userLogin?.shouldSendConnectionWarning()) {
+      const validationToken = await emailValidationDemandRepository.save(foundUser.id);
```

Comment:
@theotime2005 Il faudrait générer le token de validation uniquement si l'email n'est pas déjà validé. 
```suggestion
      const validationToken = !foundUser.emailConfirmedAt ? await emailValidationDemandRepository.save(foundUser.id) : null;
```

Et au niveau de l'email, utiliser l'URL de validation uniquement si le token est présent. Pour ça, je pense que l'idéal est de modifier la fonction `urlBuilder.getEmailValidationUrl` pour qu'elle retourne directement `redirectUrl` s'il n'y a pas de token:

```
function getEmailValidationUrl({ locale, redirectUrl, token } = {}) {
  if (!token) return redirectUrl;
```
(+ ajouter une test pour ce cas de `getEmailValidationUrl`)